test/integration_test/common_test.go

```diff
@@ -1062,6 +1093,94 @@ func IsEks() bool {
 	return false
 }
 
+// SetupTestRail checks if the required parameters for testrail are passed, verifies connectivity and creates milestone if it does not exist
+func SetupTestRail() {
+	var err error
+	testrailutils.RunName = os.Getenv(testrailRunNameVar)
+	if err == nil {
+		logrus.Infof("Testrail Run name: %s", testrailutils.RunName)
+	}
+	testrailutils.JobRunID = os.Getenv(testrailRunIDVar)
+	if err == nil {
+		logrus.Infof("Testrail Run ID: %s", testrailutils.JobRunID)
+	}
+	testrailutils.MilestoneName = os.Getenv(testrailMilestoneVar)
+	if err == nil {
+		logrus.Infof("Testrail Milestone  %s", testrailutils.MilestoneName)
+	}
+	testrailutils.JenkinsBuildURL = os.Getenv(testrailJenkinsBuildURLVar)
+	if err == nil {
+		logrus.Infof("Testrail Jenkins Build URL: %s", testrailutils.JenkinsBuildURL)
+	}
+	testrailHostname = os.Getenv(testrailHostVar)
+	if err == nil {
+		logrus.Infof("Testrail Host: %s", testrailHostname)
+	}
+	testrailUsername = os.Getenv(testrailUserNameVar)
+	if err == nil {
+		logrus.Infof("Testrail Host: %s", testrailUsername)
+	}
+	testrailPassword = os.Getenv(testrailPasswordVar)
+	if err == nil {
+		logrus.Infof("Testrail run name: %s", testrailPassword)
+	}
```

Comment:
We are checking err == nil but not assigning anything to err. We can check against the actual variables if they are empty.

Suggestion. We can do it this way if you want https://github.com/portworx/pds-agent/blob/master/cmd/server/config.go#:~:text=err%20%3A%3D%20y.Parse(args)
Another repo where this method is used to read env variable https://github.com/portworx/px-object-controller/blob/b579b19efa544385b8136fedd4c6d1a7f9d2ee0d/cmd/px-object-controller/main.go

Reply:
Makes sense, will check for the values of the actual variables. The reluctance to use the `yag` way is because we should not fail here for the testrail env variables.